### PR TITLE
Section4

### PR DESCRIPTION
**Pull request by raphtorru**
Focuses on string processing. Just 2 simple tasks. Adding commas to a numeric string and deleting all ocurrences of a char in a string.

### CODE REVIEW

File: StringProcessing.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    println("The end of task one.");
+  }
+
+  /** Adds commas to a number */
+  private String addCommasToNumericString(String digits) {
+    if (!isNumbers(digits)) {
+      return digits + " is not a number.";
+    }
+
+    if (digits.length() / 3.0 <= 1) return digits;
```
Comment (timonbimon) [inline StringProcessing.java]: this is pretty cryptic... 
why not do: digits.length <= 3

File: StringProcessing.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    String result = "";
+    int numLeading;
+    int numCommas;
+    if (digits.length() % 3 == 0) {
+      numLeading = 3;
+      numCommas = digits.length() / 3 - 1;
+    } else {
```
Comment (timonbimon) [inline StringProcessing.java]: one way to make this a little simpler would be to set 

   int   numLeading = digits.length() % 3;
    int  numCommas = digits.length() / 3;

before the if and then in the if

just do 
numLeading =3
and numCommas -= 1

then you dont need the else

File: StringProcessing.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  /** Adds number of commas to a String */
+  private String addCommas(int numCommas, String str) {
+    String result = "";
+    int i = 0;
+    while (numCommas > 0) {
+      result += ",";
+      numCommas--;
+      for (int j = 0; j < 3; j++) {
+        result += str.charAt(i + j);
```
Comment (timonbimon) [inline StringProcessing.java]: cant you use a substring instead of getting three consecutive characters?

**Comment by raphtorru** [comment]
Geil! Des wird ja automatisch geupdated hier.

**Comment by timonbimon** [comment]
haha ja ist nice :D

**Pull request merged**